### ISSUE TITLE
bug: when running both 'basic' and 'development' profile you get an JWT bean exists error

### ISSUE THREAD

**Issue by mswertz** [molgenis/molgenis-service-armadillo#481]
Motivation:
- we want to run development mode in combination with basic auth
- in the current setup you need both 'basic' and 'development'

Problem:
- two JWT beans are loaded by spring because the configs are not mutual exlusive, see discussion below

Expected:
- when running basic mode you will run the 'dummy' JWT, irrespective whether you are in 'development' mode

**Comment by mswertz** [comment]
om dit te voorkomen zouden we moeten zorgen dat die JwtDecordConfig en JwtDecoderConfigLocal nooit tegelijk aanstaan.

De JwtDecoderConfig staat aan als @ConditionalOnProperty(
    prefix = "spring.security.oauth2.resourceserver",
    value = {"jwt.issuer-uri", "opaquetoken.client-id"}) zie [https://github.com/molgenis/molgenis-service-armadillo/blob/e5f7813e032c223fe59419[…]main/java/org/molgenis/armadillo/security/JwtDecoderConfig.java](https://github.com/molgenis/molgenis-service-armadillo/blob/e5f7813e032c223fe594199abe188b703de3857e/armadillo/src/main/java/org/molgenis/armadillo/security/JwtDecoderConfig.java#L17C1-L20C15)

Dus dan zou JwtDecoderConfigLocal moeten afgaan in het andere geval. Zou dat werken als we zouden doen @ConditionalOnProperty(
    prefix = "spring.security.oauth2.resourceserver", matchIfMissing=true) ofzo? Zou het mooiste zijn want niet afhankelijk van profielen maar …. (edited) 

Alternatief is dat we de JwtDecoderConfigLocal aanpassen. Daar staat nu @Profile("basic") => hier willen we dus dat ie niet afgaat in combinatie met andere profielen met name ‘development’. Zou dit werken? @Profile("basic & !development"). Dit zou wellicht de simpelste oplossing zijn.

**Comment by clemens-tolboom** [comment]
We could use according to docs

```java
@Profile({"basic", "!development"})
```

> @Profile({"p1", "!p2"})}, registration will occur if profile 'p1' is active or if profile 'p2' is not active.

**Comment by clemens-tolboom** [comment]
Mimicking *JwtDecoderConfig* and using `matchIfMissing = true` like

```java
@ConditionalOnProperty(
    prefix = "spring.security.oauth2.resourceserver",
    value = {"jwt.issuer-uri", "opaquetoken.client-id"},
    matchIfMissing = true)
@Configuration
public class JwtDecoderConfigLocal {
```

makes me able to login using local development.

**Comment by clemens-tolboom** [comment]
Used workflow is without using web

```
curl -u admin:admin http://localhost:8080/whitelist
curl -u admin:admin -v -H 'Content-Type: multipart/form-data' -F "file=@dsBase_6.3.0-permissive.tar.gz" -X POST http://localhost:8080/install-package

docker container restart dsbaseclient_armadillo_1
sleep 120

curl -u admin:admin -X POST http://localhost:8080/whitelist/dsBase
curl -u admin:admin http://localhost:8080/whitelist
```

**Comment by clemens-tolboom** [comment]
Long story short this running locally does not work:

```
java -Dspring.profiles.active=basic,development \
 -Dspring.security.user.password=admin -jar build/libs/molgenis-armadillo-3.4.0-SNAPSHOT.jar
```

The URLs */whitelist* and */install-package* are not available for me ... is it RServer not/wrong running?

**Comment by mswertz** [comment]
Can you try http://localhost:8080/ds-profiles/{name} e.g. http://localhost:8080/ds-profiles/default

I would expect this results in a JSON containing also a whitelist property?

**Comment by clemens-tolboom** [comment]
Running 
```
curl -u admin:admin http://localhost:8080/ds-profiles/default
```
gives (piping through `json_pp`)
```json
{
   "container" : {
      "status" : "NOT_RUNNING",
      "tags" : []
   },
   "functionBlacklist" : [],
   "host" : "localhost",
   "image" : "datashield/armadillo-rserver",
   "name" : "default",
   "options" : {
      "datashield.seed" : "342325352"
   },
   "packageWhitelist" : [
      "dsBase",
      "x"
   ],
   "port" : 6311
}
```

**Comment by clemens-tolboom** [comment]
## Warning

When requesting for `/ds-profiles/default` a warning is logged. Not sure this is a bug on our behalve.

```
11:56:51.794 [http-nio-8080-exec-9|3420C1E9] WARN  o.m.armadillo.profile.DockerService - Couldn't inspect image
com.github.dockerjava.api.exception.NotFoundException: Status 404: No such image: default:latest
	at com.github.dockerjava.jaxrs.filter.ResponseStatusExceptionFilter.filter(ResponseStatusExceptionFilter.java:62)
```

**Comment by StuartWheater** [comment]
Checked out source, and building...

**Comment by clemens-tolboom** [comment]
### Develop mode

Running in development mode

```zsh
java \
  -Dspring.profiles.active=development \
  -Dspring.config.additional-location=file:armadillo/src/main/resources/application-local.yml \
  -jar build/libs/molgenis-armadillo-3.4.0-SNAPSHOT.jar
```

### OpenAPI / Swagger

I can access http://localhost:8080/swagger-ui/index.html#/developer/installPackage logged in using basic auth.

Test run through OpenAPI we obtain a curl example.

### Curl

```zsh
curl -X 'POST' \
  'http://localhost:8080/install-package' \
  -H 'accept: */*' \
  -H 'Content-Type: multipart/form-data' \
  -F 'file=@clemens.png;type=image/png'
  ```

**Comment by clemens-tolboom** [comment]
This is AFAIK resolved through the new release v3.4.0

Does it makes sense we keep the curl commands in the docs?

**Comment by timcadman** [comment]
We don't use profiles any more
